invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js

```diff
@@ -5,32 +5,136 @@
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import PropTypes from "prop-types";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { i18next } from "@translations/invenio_administration/i18next";
 import CopyButton from "./CopyButton";
-import { Table, Header } from "semantic-ui-react";
+import { Table, Header, Dropdown, Grid } from "semantic-ui-react";
+import { http } from "@js/invenio_administration/src/api/config.js";
+import { capitalize } from "lodash";
+
+/** Map of known formats and their name. */
+const knownFormats = {
+  "oai_dc": "OAI Dublin Core",
+  "datacite": "DataCite",
+  "oai_datacite": "OAI DataCite",
+}
 
 function LinksTable({ data }) {
+  const [links, setLinks] = useState(data.links);
+  const [formats, setFormats] = useState([]);
+
+  useEffect(async () => {
+    try {
+      const response = await http.get(
+        '/api/oaipmh/formats'
+      );
+      const formats = response.data?.hits?.hits;
+      if (Array.isArray(formats) && formats.length > 0) {
+        const serialized = formats.map((formt) => {
+          return {
+            key: formt.id,
+            value: formt.id,
+            text: knownFormats[formt.id] ?? formatKeyToName(formt.id)
+          }
+        });
+        setFormats(serialized);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
```

Comment:
Empty dependencies means that this hook is only used once, when the component is first rendered.

This hook will retrieve OAI formats from the backend and store them in the components state.

Reply:
The fact that we have to add a comment in the PR to explain this makes me very scared about future changes/refactorings, especially because we don't have frontend integration tests. Please consider in the future to avoid using `useEffect` or, eventually, we need to make it very clear and readable what this code is doing and why. :)